### DISCUSSION TITLE
Transitive licensing obligations through dependencies

### DISCUSSION THREAD

**Discussion by artemonster** [tauri-apps/tauri#5561] [Q&A] [answered]
Hi,

sorry if this is a completely silly question, probably caused by my complete misunderstanding of the topic. While tauri is licensed with MIT/Apache, according to FOSSA, there are some transitive dependencies for tauri that obligate you to provide source code (under Obligations, "Disclose source for projects" part), i.e.
rug (1.17.0) licensed with GPL2.0:
“If you distribute this software in an executable, you must make your source code available by providing it alongside your distribution or listing an offer, good for 3 years, to obtain the code. ”
Does this mean the complete source code of the whole application built with tauri or just the library/dependency itself? 
Thank you

**Comment by nothingismagick** [comment, marked as answer]
Thanks for the question. and it is not silly. I am maintaining our license complaince, and we take it very seriously. Please note, that rug is a dev dependency, and not shipped with your application.  

![image](https://user-images.githubusercontent.com/35242872/200194539-0d7b805a-15cb-4d73-8894-f1f46f725c46.png)

**Reply by artemonster** [reply]
Thanks for pointing this out! While we're on the topic, may I ask if AppImage is chosen as a distribution method, how do you include licence notices for all the dependencies that require it? Do you put it inside the archive and be done with it, or it should be linked inside the application somehow (i.e. under Help/About section?)

**Reply by nothingismagick** [reply]
Well, shipping all the licenses would very clearly bloat the final binary. Some projects have a link to a website that lists all dependencies and their licenses, others share a fossa or similar page. The only problem with fossa, in my experience, is that it is full of false negatives. Even writing the word "GPL" anywhere in code is enough to flag their system...
